utils/wasm-gen/src/config/module.rs

```diff
@@ -0,0 +1,364 @@
+            max_values,
+            memory64_enabled,
+            memory_max_size_required,
+            memory_offset_choices,
+            min_data_segments,
+            min_element_segments,
+            min_elements,
+            min_exports,
+            min_funcs,
+            min_globals,
+            min_imports,
+            min_memories,
+            min_tables,
+            min_tags,
+            min_types,
+            min_uleb_size,
+            multi_value_enabled,
+            reference_types_enabled,
+            relaxed_simd_enabled,
+            saturating_float_to_int_enabled,
+            sign_extension_enabled,
+            simd_enabled,
+            float_enabled,
+            threads_enabled,
+            allowed_instructions,
+            max_table_elements,
+            table_max_size_required,
+            memory_grow_enabled,
+            call_indirect_enabled,
+        })
+    }
+}
+
+/// Arbitrary wasm module generation params.
+///
+/// These are params that are allowed to be randomly set.
+/// All of them are later used to instantiate `wasm_smith::SwarmConfig`.
+#[derive(Debug, Clone)]
+pub struct ArbitraryParams {
+    available_imports: Option<Vec<u8>>,
+    canonicalize_nans: bool,
+    export_everything: bool,
+    max_aliases: usize,
+    max_components: usize,
+    max_data_segments: usize,
+    max_element_segments: usize,
+    max_elements: usize,
+    max_globals: usize,
+    max_instances: usize,
+    max_modules: usize,
+    max_nesting_depth: usize,
+    max_tags: usize,
+    max_type_size: u32,
+    max_types: usize,
+    max_values: usize,
+    memory_max_size_required: bool,
+    memory_offset_choices: (u32, u32, u32),
+    min_data_segments: usize,
+    min_element_segments: usize,
+    min_elements: usize,
+    min_globals: usize,
+    min_tables: u32,
+    min_tags: usize,
+    min_types: usize,
+    min_uleb_size: u8,
+    threads_enabled: bool,
+    max_table_elements: u32,
+    table_max_size_required: bool,
+    max_memory_pages: u64,
+}
+
+impl Arbitrary<'_> for ArbitraryParams {
+    fn arbitrary(u: &mut Unstructured<'_>) -> Result<Self> {
+        let random_swarm = u.arbitrary()?;
+        let SwarmConfig {
+            available_imports,
+            canonicalize_nans,
+            export_everything,
+            max_aliases,
+            max_components,
+            max_data_segments,
+            max_element_segments,
+            max_elements,
+            max_globals,
+            max_instances,
+            max_modules,
+            max_nesting_depth,
+            max_tags,
+            max_type_size,
+            max_types,
+            max_values,
+            memory_max_size_required,
+            memory_offset_choices,
+            min_data_segments,
+            min_element_segments,
+            min_elements,
+            min_globals,
+            min_tables,
+            min_tags,
+            min_types,
+            min_uleb_size,
+            threads_enabled,
```

Comment:
As I can see these params are going to be used in SwarmConfig. So, for example threads_enabled will be enabled 50/50. Whether this can lead to more often invalid wasm codes generating ?

Reply:
Will test `threads_enabled` in next PRs as testing whether it makes gear wasm invalid or not is quite hard with current code base. Better after #2982